packages/eas-cli/src/project/publish.ts

```diff
@@ -200,43 +206,62 @@ export function loadMetadata(distRoot: string): Metadata {
   if (metadata.bundler !== 'metro') {
     throw new Error('Only bundles created with Metro are currently supported');
   }
+  const platforms = Object.keys(metadata.fileMetadata);
+  if (platforms.length === 0) {
+    Log.warn('No updates were exported for any platform');
+  }
+  Log.debug(`Loaded ${platforms.length} platform(s): ${platforms.join(', ')}`);
   return metadata;
 }
 
-export async function collectAssetsAsync({
-  inputDir,
-  platforms,
-}: {
-  inputDir: string;
-  platforms: PublishPlatform[];
-}): Promise<CollectedAssets> {
-  const distRoot = await resolveInputDirectoryAsync(inputDir);
-  const metadata = loadMetadata(distRoot);
-
-  const assetsFinal: CollectedAssets = {};
-  for (const platform of platforms) {
-    assetsFinal[platform] = {
+export function filterExportedPlatformsByFlag<T extends Partial<Record<Platform, any>>>(
+  record: T,
+  platformFlag: ExpoCLIExportPlatformFlag
+): T {
+  if (platformFlag === 'all') {
+    return record;
+  }
+
+  const platform = platformFlag as Platform;
+
+  if (!record[platform]) {
+    throw new Error(
+      `--platform="${platform}" not found in metadata.json. Available platform(s): ${Object.keys(
```

Comment:
IMO these error messages sound as if `--platform="${platform}"` (and `--input-dir="${inputDir}"`) was sth you could find in metadata.json. I mean the exact string.

Reply:
Thought about this a bit more and the entire error message is generally not helpful. User's probably don't even know what a `metadata.json` is since we don't document it and the format is non-standard. 

<img width="824" alt="Screen Shot 2022-11-08 at 11 39 02 AM" src="https://user-images.githubusercontent.com/9664363/200636638-16fd66b0-c17a-422c-ab9e-a6de521c4cd9.png">

All they do know is that the value they provided `--platform="..."` isn't available.